### DISCUSSION TITLE
Support: <tab> to select first item from the list

### DISCUSSION THREAD

**Discussion by rajeshisnepali** [neoclide/coc.nvim#4255] [Q&A] [answered]
I remember in the older versions, I could hit <tab> to select (autocomplete) the first item and then the second, and so on. 

But in the newer version, pressing <tab> selects the second item, and to select the first item, I've to do either of the following options
- hit `enter` to autocomplete
- `tab` and then `shift+tab` to autocomplete.

![2022-10-04_15-06](https://user-images.githubusercontent.com/23610013/193784215-9435a3f8-33e3-44f7-9b75-dc6feba20c21.png)

**Nvim** 
```
NVIM v0.8.0-1210-gd367ed9b2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by runner@fv-az505-656
```

**My configuration**
```
" use <tab> for trigger completion and navigate to the next complete item
function! CheckBackspace() abort
  let col = col('.') - 1
  return !col || getline('.')[col - 1]  =~# '\s'
endfunction

inoremap <silent><expr> <Tab>
      \ coc#pum#visible() ? coc#pum#next(1) :
      \ CheckBackspace() ? "\<Tab>" :
      \ coc#refresh()

inoremap <expr> <Tab> coc#pum#visible() ? coc#pum#next(1) : "\<Tab>"
inoremap <expr> <S-Tab> coc#pum#visible() ? coc#pum#prev(1) : "\<S-Tab>"
inoremap <silent><expr> <cr> coc#pum#visible() ? coc#_select_confirm() : "\<C-g>u\<CR>"

```

**Comment by fannheyward** [comment, marked as answer]
Try `suggest.noselect` to true.

**Reply by rajeshisnepali** [reply]
I add the above params in  `~/.config/nvim/coc-settings.json`

```
{    
    "suggest.noselect": true    
}  
```

**Comment by chemzqm** [comment]
This is my `<tab>`

```
    inoremap <silent><expr> <TAB>
          \ coc#pum#visible() ? coc#pum#insert():
          \ <SID>check_back_space() ? "\<Tab>" :
          \ coc#refresh()
```
Which insert text of selected item when pum is visible.
The behavior changed from coc.nvim 0.0.82 to make it more like VSCode, and some features just work.
